### PR TITLE
Fix PrefetchingResultSetIterator regression introduced in latest commit

### PR DESCRIPTION
**Pull request by eappere**
Latest commit introduces a regression & doesn't handle all corner cases. Fix this by calling maybePrefetch in constructor & update unit tests accordingly.

### CODE REVIEW

File: connector/src/main/scala/com/datastax/spark/connector/rdd/reader/PrefetchingResultSetIterator.scala [hunk 1]
```diff
@@ -51,8 +53,11 @@ class PrefetchingResultSetIterator(resultSet: AsyncResultSet, timer: Option[Time
...
 
   override def next(): Row = {
-    maybePrefetch()
     val row = currentIterator.next()
+    maybePrefetch()
     row
   }
+
+  // See comment in maybePrefetch
+  maybePrefetch()
```
Comment (damsallem) [inline connector/src/main/scala/com/datastax/spark/connector/rdd/reader/PrefetchingResultSetIterator.scala:65]: I don't get why it fix anything, since you were calling maybePrefetch() in the next() method before the call to currentIterator.next().
Thus, even if the first page was empty, then the next having elements would have been prefetch.
Are you sure your UTs is failing without this change ?

**Review by damsallem — CHANGES_REQUESTED** [review]

File: connector/src/main/scala/com/datastax/spark/connector/rdd/reader/PrefetchingResultSetIterator.scala (hunk 1 above)
Comment (eappere) [reply connector/src/main/scala/com/datastax/spark/connector/rdd/reader/PrefetchingResultSetIterator.scala:65]: Yes, it is possible to have results made of pages with zero elements only

**Review by damsallem — APPROVED** [review]
LGTM
